### DISCUSSION TITLE
How do you add error handling to glowroot central 13.4?

### DISCUSSION THREAD

**Discussion by YamasakiRhys** [glowroot/glowroot#829] [Q&A] [answered]
We are currently using glowroot version 13.4 and some error detection changes are need to be made for security reasons.
I tried adding the required error tags to web.xml in glowroot central, but the jar isn't able to start up after being built.
Is there a way to properly implement error handling?

**Comment by Never-lab** [comment, marked as answer]
Don't inject Tomcat-style `<error-page>` into Central's packaging — that is why the rebuilt jar fails to start. Central is not a WAR you harden with scanner "web.xml remediations."

Use a current [0.14.x](https://github.com/glowroot/glowroot/releases) Central, put it behind auth / TLS / network controls, and leave the dist alone. Same pattern as issue #827.

If you still see a specific unexpected 500 on a **valid** admin action on current Central, open a new issue with version + steps.
